### DISCUSSION TITLE
User creation

### DISCUSSION THREAD

**Discussion by dkoci** [quay/quay#1004] [Q&A]
Hi.
I disabled user creation for Non-Superusers. It means users can only be created in the superuser tool.

![image](https://user-images.githubusercontent.com/5565897/144241053-1c03a9b2-e391-45a1-81e5-09ea4dc4eb2b.png)

Is superuser tool another tool which has to be installed or it is "Superuser admin panel"? I am superuser but I dont see any option for user creation.

![image](https://user-images.githubusercontent.com/5565897/144241939-ac907d18-c4a1-42f5-9438-61ad67857bc8.png)

**Comment by flavianmissi** [comment]
Hi @dkoci, do you still need help with this?

**Reply by dkoci** [reply]
I would like to prevent normal users to create organizations. Only admin users. Is that possible?

**Reply by flavianmissi** [reply]
not yet @dkoci but we have a feature request for that, which you can follow: https://issues.redhat.com/browse/PROJQUAY-402

**Reply by dkoci** [reply]
That would be really nice feature. Hope it will be available soon.

**Reply by dkoci** [reply]
Anything new here? Will be feature which prevents normal users from creating organizations available in 3.8?

**Reply by HammerMeetNail** [reply]
Hey @dkoci, there is a new feature in Quay that can restrict users from creating organizations in 3.8. The flag is `FEATURE_RESTRICTED_USERS` and when enabled will prevent all users from doing just about anything in any org they haven't been invited to. We put together a demo here: https://www.youtube.com/watch?v=lzP0msIhK_o

We'll cut the upstream release tomorrow.

**Reply by dkoci** [reply]
Nice to hear that. Thank you.

**Reply by dkoci** [reply]
@HammerMeetNail I tried a new feature that restrits users from creating organizations but I found something which I think that is not ok. Every time when user log in for the very first time there is organization created which has username as the name of organization. But in case of `FEATURE_RESTRICTED_USERS=True` user cant create repositories in this auto created organization. I think this is wrong. User should be able to create repositories in organization which is auto created after first log in (despite FEATURE_RESTRICTED_USERS option is enabled).

**Reply by dmesser** [reply]
This is intentional @dkoci. The goal of `FEATURE_RESTRICTED_USERS` is that users can only create content where they are supposed to and where there a certain checks and balances in place. Otherwise you end up with container image sprawl. If we'd allowed restricted users to still create repos and content in their own account (the "organization" you mentioned that has their name) they would effectively circumvent this control and registry teams would still face unrestricted content growth.

**Reply by dkoci** [reply]
Is there any option that admins can give users permission to create repos in their own organization?

**Reply by dmesser** [reply]
@dkoci not at the moment. Can you describe a scenario where this is needed but the users aren't allowed to create their own organizations?

**Reply by dkoci** [reply]
We have a rule in our company that users are not allowed to create organizations. Only admins can do that (the same goes for gitlab projects. Only admins can create it). But it is allowed users to create repositories in organization.

**Reply by dmesser** [reply]
What are you trying to achieve with that?

**Reply by dkoci** [reply]
We dont want there would be unnecessary organizations which users could create. We also want that every organization is properly named.
